impala/hiveserver2.py

```diff
@@ -1112,37 +1143,43 @@ def __init__(self, service, handle, config, hs2_protocol_version,
 
     def close(self):
         req = TCloseSessionReq(sessionHandle=self.handle)
-        self._rpc('CloseSession', req)
+        # CloseSession rpcs don't need retries since we catch all exceptions and close
+        # transport.
+        self._rpc('CloseSession', req, False)
 
     def execute(self, statement, configuration=None, run_async=False):
         req = TExecuteStatementReq(sessionHandle=self.handle,
                                    statement=statement,
                                    confOverlay=configuration,
                                    runAsync=run_async)
-        return self._operation('ExecuteStatement', req)
+        # Do not try to retry http requests.
+        # Read queries should be idempotent but most dml queries are not. Also retrying
+        # query execution from client could be expensive and so likely makes sense to do
+        # it if server is also aware of the retries.
+        return self._operation('ExecuteStatement', req, False)
```

Comment:
Thanks for explicitly passing a **False** param even though the default value is False for retries. Improves the readability.